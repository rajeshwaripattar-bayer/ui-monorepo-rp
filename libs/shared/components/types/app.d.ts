export interface Notification {
  open: boolean
  timeout?: number
  message: string
  dismissButtonLabel?: string
  actionButtonProps?: {
    label: string
    onAction: () => void
  }
}
