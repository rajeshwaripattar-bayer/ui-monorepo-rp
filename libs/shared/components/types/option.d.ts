export type Option = { value: string; text: string }
export type ApplyAllProps = { label?: string; applied?: boolean }
