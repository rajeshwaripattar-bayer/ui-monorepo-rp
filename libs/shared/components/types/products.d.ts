import { DiscDiscount, PriceValueAndCurrency } from '@gc/types'
import { Strategy } from './discounts'

export interface ProductWithPrice {
  canView?: boolean
  canOrder?: boolean
  cropCode: string
  cropName: string
  name: string
  code: string
  quantity: number
  netQuantity: number
  totalDiscount: number
  discounts?: { programName: string; strategy: Strategy }[]
  discountedUnitPrice: number
  unitPrice: number
  subTotal: number
  warehouse: { value: string; text: string }
  statusCounts?: {
    status: string
    count: number
    isBold?: boolean
  }[]
}

export interface PriceAndDiscountSummary {
  grossPrice: number
  discount: number
  discountPercentage: number
  discounts: {
    name: string
    value: number
  }[]
  netPrice: number
  discretionaryDiscounts: DiscDiscount[]
  totalDiscretionaryDiscount: PriceValueAndCurrency
  grossPriceDiscretionaryDiscount: PriceValueAndCurrency
  netPriceDiscretionaryDiscount: PriceValueAndCurrency
}

export interface ProductsByCrop {
  crop: string
  products: ProductWithPrice[]
  avgPrice: number
  summary: PriceAndDiscountSummary
}

export interface QuantityUpdateRequest {
  quantity: string | number
  productWithPrice: ProductWithPrice
  onlyCache?: boolean
  contingencyType?: 'alert' | 'dialog'
}

export interface ReturnedProducts {
  cropCode: string
  cropName: string
  name: string
  code: string
  quantity: number
  warehouseName?: string
  UOM: string
  masterOrderNumber?: string
}
export interface ReturnedProductsByCrop {
  crop: string
  products: ReturnedProducts[]
}

export type StockOrderProduct = {
  name: string
  warehouse: { value: string; text: string }
  quantity: number
  confirmedQuantity: number
  unconfirmedQuantity: number
}

export type StockOrderProductsByCrop = {
  crop: string
  count: number
  products: StockOrderProduct[]
}
