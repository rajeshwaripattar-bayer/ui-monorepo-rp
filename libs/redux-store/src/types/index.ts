import type { ExtractReducerState, EnhancedStore } from '@gc/types'
import { ccApi, middlewareApi, fgCcApi } from '../rtk-apis'
import type { AppState, CartState, FiltersState } from '../slices'

export type GlobalRootState = {
  app: AppState
  cart: CartState
  filters: FiltersState
  [ccApi.reducerPath]: ExtractReducerState<typeof ccApi.reducer>
  [fgCcApi.reducerPath]: ExtractReducerState<typeof fgCcApi.reducer>
  [middlewareApi.reducerPath]: ExtractReducerState<typeof middlewareApi.reducer>
}

export type GlobalAppStore = EnhancedStore<GlobalRootState>
export type GlobalAppDispatch = GlobalAppStore['dispatch']
