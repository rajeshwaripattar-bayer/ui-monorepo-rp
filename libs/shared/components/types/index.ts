export * from './app'
export * from './option'
export * from './products'
