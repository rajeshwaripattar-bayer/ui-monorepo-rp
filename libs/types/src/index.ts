export * from './accounts'
export * from './aem'
export * from './shared'
export * from './orderDetails'
export * from './stockOrder'
export * from './api'
export * from './app'
export * from './cart'
export * from './ccApi'
export * from './configData'
export * from './discounts'
export * from './domainDef'
export * from './farmers'
export * from './helpers'
export * from './locale'
export * from './orders'
export * from './products'
export * from './programs'
export * from './quoteDetails'
export * from './quotes'
export * from './redux'
export * from './deliveries'
export * from './selectedAccount'
export * from './user'
export * from './filter'
export * from './shipmentDeliveries'
export * from './fgCcApi'
export * from './addressInfo'

