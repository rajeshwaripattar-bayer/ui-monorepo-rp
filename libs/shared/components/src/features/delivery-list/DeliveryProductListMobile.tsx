import styles from './DeliveryProductListMobile.module.scss'
import { Card, CardContent, CardTitle, CardBody } from '@element/react-card'
import { List } from '../../ui-common/list/List'
import { TypoCaption, TypoSubtitle } from '@element/react-typography'
import { useTranslation } from 'react-i18next'
import { Delivery, DeliveryEntry } from '@gc/types'
import { interpunct } from '@gc/constants'

export interface DeliveryProductListMobileProps {
  delivery: Delivery
}

export function DeliveryProductListMobile(props: DeliveryProductListMobileProps) {
  const { t } = useTranslation()

  const getStyledListItems = (entries: Array<DeliveryEntry>) => {
    const items: Array<object> = []
    entries.map((deliveryEntry: DeliveryEntry) =>
      items.push({
        code: deliveryEntry.deliveryItemNumber,
        trailingBlock: (
          <TypoCaption>{`${deliveryEntry.quantity} ${deliveryEntry.salesUnitOfMeasureCode ?? ''}`}</TypoCaption>
        ),
        primaryText: deliveryEntry.product.name,
        secondaryText: (
          <>
            {deliveryEntry.batchName && <TypoCaption data-testid='batchName'>{deliveryEntry.batchName}</TypoCaption>}
            {deliveryEntry.batchName && deliveryEntry.seedSize && interpunct}
            {deliveryEntry.seedSize && <TypoCaption data-testid='seedSize'>{deliveryEntry.seedSize}</TypoCaption>}
          </>
        )
      })
    )
    return items
  }

  return (
    <Card variant='outlined' className={styles.card}>
      <CardContent>
        <CardTitle
          data-testid='title'
          primaryText={<TypoSubtitle level={1}>{`${t('deliveries.products_in_delivery.label')}`}</TypoSubtitle>}
        />
        <CardBody data-testid='orderId' typographyType='overline' className={styles.card_body}>{`${t(
          'orders.order.label'
        )} ${
          props.delivery.salesOrderId.indexOf('-') > 0
            ? props.delivery.salesOrderId.split('-')[1]
            : props.delivery.salesOrderId
        }`}</CardBody>
        <List
          data-testid='productList'
          className={styles.product_list_mobile}
          items={getStyledListItems(props.delivery.entries)}
          noHover={true}
          divider
        />
      </CardContent>
    </Card>
  )
}

export default DeliveryProductListMobile
