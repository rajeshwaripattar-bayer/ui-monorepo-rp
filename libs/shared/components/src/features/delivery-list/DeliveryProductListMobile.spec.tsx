import { render } from '@testing-library/react'
import { Delivery } from '@gc/types'

import DeliveryProductListMobile from './DeliveryProductListMobile'

describe('DeliveryProductListMobile', () => {
  it('should render (snapshot test) successfully', () => {
    const delivery: Delivery = {
      agent: {
        name: 'BRIAN HAVLIK',
        uid: '0009146406'
      },
      code: '0801760288',
      createdOnDateTime: '2024-06-24T13:23:14-05:00',
      entries: [
        {
          deliveryItemNumber: '000020',
          product: {
            available: 0.0,
            code: '000000000091057659',
            name: 'PR108-20RIB VT2PRIB 80M BAG ELT500 N-B-E'
          },
          quantity: 1,
          salesOrderEntryNumber: '20',
          salesUnitOfMeasureCode: 'ST'
        },
        {
          batchName: 'L42S7RMKP',
          deliveryItemNumber: '000010',
          product: {
            available: 0.0,
            code: '000000000088333144',
            name: '189-64VT2PRIB 50U BOX BAS500 B'
          },
          quantity: 1,
          salesOrderEntryNumber: '10',
          salesUnitOfMeasureCode: 'SSU',
          seedSize: 'R2'
        }
      ],
      farmer: {
        name: 'CARL GREEN',
        uid: '0009221009'
      },
      salesOrderId: 'BC-0002222577',
      salesYear: '2024',
      status: 'DELIVERED'
    }
    const { baseElement } = render(<DeliveryProductListMobile delivery={delivery} />)
    expect(baseElement).toMatchSnapshot()
  })

  it('should render title successfully', () => {
    const delivery: Delivery = {
      agent: {
        name: 'BRIAN HAVLIK',
        uid: '0009146406'
      },
      code: '0801760288',
      createdOnDateTime: '2024-06-24T13:23:14-05:00',
      entries: [
        {
          deliveryItemNumber: '000020',
          product: {
            available: 0.0,
            code: '000000000091057659',
            name: 'PR108-20RIB VT2PRIB 80M BAG ELT500 N-B-E'
          },
          quantity: 1,
          salesOrderEntryNumber: '20',
          salesUnitOfMeasureCode: 'ST'
        },
        {
          batchName: 'L42S7RMKP',
          deliveryItemNumber: '000010',
          product: {
            available: 0.0,
            code: '000000000088333144',
            name: '189-64VT2PRIB 50U BOX BAS500 B'
          },
          quantity: 1,
          salesOrderEntryNumber: '10',
          salesUnitOfMeasureCode: 'SSU',
          seedSize: 'R2'
        }
      ],
      farmer: {
        name: 'CARL GREEN',
        uid: '0009221009'
      },
      salesOrderId: 'BC-0002222577',
      salesYear: '2024',
      status: 'DELIVERED'
    }
    const { getByTestId, getByText } = render(<DeliveryProductListMobile delivery={delivery} />)
    expect(getByTestId('title')).toBeTruthy()
    expect(getByText('deliveries.products_in_delivery.label')).toBeTruthy()
  })

  it('should render sub title (order Id) successfully', () => {
    const delivery: Delivery = {
      agent: {
        name: 'BRIAN HAVLIK',
        uid: '0009146406'
      },
      code: '0801760288',
      createdOnDateTime: '2024-06-24T13:23:14-05:00',
      entries: [
        {
          deliveryItemNumber: '000020',
          product: {
            available: 0.0,
            code: '000000000091057659',
            name: 'PR108-20RIB VT2PRIB 80M BAG ELT500 N-B-E'
          },
          quantity: 1,
          salesOrderEntryNumber: '20',
          salesUnitOfMeasureCode: 'ST'
        },
        {
          batchName: 'L42S7RMKP',
          deliveryItemNumber: '000010',
          product: {
            available: 0.0,
            code: '000000000088333144',
            name: '189-64VT2PRIB 50U BOX BAS500 B'
          },
          quantity: 1,
          salesOrderEntryNumber: '10',
          salesUnitOfMeasureCode: 'SSU',
          seedSize: 'R2'
        }
      ],
      farmer: {
        name: 'CARL GREEN',
        uid: '0009221009'
      },
      salesOrderId: 'BC-0002222577',
      salesYear: '2024',
      status: 'DELIVERED'
    }
    const { getByTestId, getByText } = render(<DeliveryProductListMobile delivery={delivery} />)
    expect(getByTestId('orderId')).toBeTruthy()
    expect(getByText('orders.order.label 0002222577')).toBeTruthy()
  })

  it('should render product list successfully', () => {
    const delivery: Delivery = {
      agent: {
        name: 'BRIAN HAVLIK',
        uid: '0009146406'
      },
      code: '0801760288',
      createdOnDateTime: '2024-06-24T13:23:14-05:00',
      entries: [
        {
          deliveryItemNumber: '000020',
          product: {
            available: 0.0,
            code: '000000000091057659',
            name: 'PR108-20RIB VT2PRIB 80M BAG ELT500 N-B-E'
          },
          quantity: 1,
          salesOrderEntryNumber: '20',
          salesUnitOfMeasureCode: 'ST'
        },
        {
          batchName: 'L42S7RMKP',
          deliveryItemNumber: '000010',
          product: {
            available: 0.0,
            code: '000000000088333144',
            name: '189-64VT2PRIB 50U BOX BAS500 B'
          },
          quantity: 1,
          salesOrderEntryNumber: '10',
          salesUnitOfMeasureCode: 'SSU',
          seedSize: 'R2'
        }
      ],
      farmer: {
        name: 'CARL GREEN',
        uid: '0009221009'
      },
      salesOrderId: 'BC-0002222577',
      salesYear: '2024',
      status: 'DELIVERED'
    }
    const { getByText } = render(<DeliveryProductListMobile delivery={delivery} />)

    expect(getByText('PR108-20RIB VT2PRIB 80M BAG ELT500 N-B-E')).toBeTruthy()
    expect(getByText('189-64VT2PRIB 50U BOX BAS500 B')).toBeTruthy()
  })

  it('should render product batch name  successfully', () => {
    const delivery: Delivery = {
      agent: {
        name: 'BRIAN HAVLIK',
        uid: '0009146406'
      },
      code: '0801760288',
      createdOnDateTime: '2024-06-24T13:23:14-05:00',
      entries: [
        {
          deliveryItemNumber: '000020',
          product: {
            available: 0.0,
            code: '000000000091057659',
            name: 'PR108-20RIB VT2PRIB 80M BAG ELT500 N-B-E'
          },
          quantity: 1,
          salesOrderEntryNumber: '20',
          salesUnitOfMeasureCode: 'SSU'
        },
        {
          batchName: 'L42S7RMKP',
          deliveryItemNumber: '000010',
          product: {
            available: 0.0,
            code: '000000000088333144',
            name: '189-64VT2PRIB 50U BOX BAS500 B'
          },
          quantity: 10,
          salesOrderEntryNumber: '10',
          salesUnitOfMeasureCode: 'SSU',
          seedSize: 'R2'
        }
      ],
      farmer: {
        name: 'CARL GREEN',
        uid: '0009221009'
      },
      salesOrderId: 'BC-0002222577',
      salesYear: '2024',
      status: 'DELIVERED'
    }
    const { getByText, getByTestId } = render(<DeliveryProductListMobile delivery={delivery} />)

    expect(getByTestId('batchName')).toBeTruthy()
    expect(getByText('L42S7RMKP')).toBeTruthy()
  })

  it('should not render product batch name, if not available', () => {
    const delivery: Delivery = {
      agent: {
        name: 'BRIAN HAVLIK',
        uid: '0009146406'
      },
      code: '0801760288',
      createdOnDateTime: '2024-06-24T13:23:14-05:00',
      entries: [
        {
          deliveryItemNumber: '000020',
          product: {
            available: 0.0,
            code: '000000000091057659',
            name: 'PR108-20RIB VT2PRIB 80M BAG ELT500 N-B-E'
          },
          quantity: 1,
          salesOrderEntryNumber: '20',
          salesUnitOfMeasureCode: 'SSU'
        },
        {
          deliveryItemNumber: '000010',
          product: {
            available: 0.0,
            code: '000000000088333144',
            name: '189-64VT2PRIB 50U BOX BAS500 B'
          },
          quantity: 10,
          salesOrderEntryNumber: '10',
          salesUnitOfMeasureCode: 'SSU',
          seedSize: 'R2'
        }
      ],
      farmer: {
        name: 'CARL GREEN',
        uid: '0009221009'
      },
      salesOrderId: 'BC-0002222577',
      salesYear: '2024',
      status: 'DELIVERED'
    }
    const { queryByTestId } = render(<DeliveryProductListMobile delivery={delivery} />)

    expect(queryByTestId('batchName')).toBeNull()
  })

  it('should render product seedsize  successfully', () => {
    const delivery: Delivery = {
      agent: {
        name: 'BRIAN HAVLIK',
        uid: '0009146406'
      },
      code: '0801760288',
      createdOnDateTime: '2024-06-24T13:23:14-05:00',
      entries: [
        {
          deliveryItemNumber: '000020',
          product: {
            available: 0.0,
            code: '000000000091057659',
            name: 'PR108-20RIB VT2PRIB 80M BAG ELT500 N-B-E'
          },
          quantity: 1,
          salesOrderEntryNumber: '20',
          salesUnitOfMeasureCode: 'SSU'
        },
        {
          batchName: 'L42S7RMKP',
          deliveryItemNumber: '000010',
          product: {
            available: 0.0,
            code: '000000000088333144',
            name: '189-64VT2PRIB 50U BOX BAS500 B'
          },
          quantity: 10,
          salesOrderEntryNumber: '10',
          salesUnitOfMeasureCode: 'SSU',
          seedSize: 'R2'
        }
      ],
      farmer: {
        name: 'CARL GREEN',
        uid: '0009221009'
      },
      salesOrderId: 'BC-0002222577',
      salesYear: '2024',
      status: 'DELIVERED'
    }
    const { getByTestId } = render(<DeliveryProductListMobile delivery={delivery} />)

    expect(getByTestId('seedSize')).toBeTruthy()
  })

  it('should not render product seedsize, if not available', () => {
    const delivery: Delivery = {
      agent: {
        name: 'BRIAN HAVLIK',
        uid: '0009146406'
      },
      code: '0801760288',
      createdOnDateTime: '2024-06-24T13:23:14-05:00',
      entries: [
        {
          deliveryItemNumber: '000020',
          product: {
            available: 0.0,
            code: '000000000091057659',
            name: 'PR108-20RIB VT2PRIB 80M BAG ELT500 N-B-E'
          },
          quantity: 1,
          salesOrderEntryNumber: '20',
          salesUnitOfMeasureCode: 'SSU'
        },
        {
          batchName: 'L42S7RMKP',
          deliveryItemNumber: '000010',
          product: {
            available: 0.0,
            code: '000000000088333144',
            name: '189-64VT2PRIB 50U BOX BAS500 B'
          },
          quantity: 10,
          salesOrderEntryNumber: '10',
          salesUnitOfMeasureCode: 'SSU'
        }
      ],
      farmer: {
        name: 'CARL GREEN',
        uid: '0009221009'
      },
      salesOrderId: 'BC-0002222577',
      salesYear: '2024',
      status: 'DELIVERED'
    }
    const { queryByTestId } = render(<DeliveryProductListMobile delivery={delivery} />)

    expect(queryByTestId('seedSize')).toBeNull
  })

  it('should render interpunct if both batchname & seedsize are available', () => {
    const delivery: Delivery = {
      agent: {
        name: 'BRIAN HAVLIK',
        uid: '0009146406'
      },
      code: '0801760288',
      createdOnDateTime: '2024-06-24T13:23:14-05:00',
      entries: [
        {
          deliveryItemNumber: '000020',
          product: {
            available: 0.0,
            code: '000000000091057659',
            name: 'PR108-20RIB VT2PRIB 80M BAG ELT500 N-B-E'
          },
          quantity: 1,
          salesOrderEntryNumber: '20',
          salesUnitOfMeasureCode: 'SSU'
        },
        {
          batchName: 'L42S7RMKP',
          deliveryItemNumber: '000010',
          product: {
            available: 0.0,
            code: '000000000088333144',
            name: '189-64VT2PRIB 50U BOX BAS500 B'
          },
          quantity: 10,
          salesOrderEntryNumber: '10',
          salesUnitOfMeasureCode: 'SSU',
          seedSize: 'R2'
        }
      ],
      farmer: {
        name: 'CARL GREEN',
        uid: '0009221009'
      },
      salesOrderId: 'BC-0002222577',
      salesYear: '2024',
      status: 'DELIVERED'
    }
    const { getByText } = render(<DeliveryProductListMobile delivery={delivery} />)

    expect(getByText('•')).toBeTruthy()
  })

  it('should not render interpunct if seedsize is not  available', () => {
    const delivery: Delivery = {
      agent: {
        name: 'BRIAN HAVLIK',
        uid: '0009146406'
      },
      code: '0801760288',
      createdOnDateTime: '2024-06-24T13:23:14-05:00',
      entries: [
        {
          deliveryItemNumber: '000020',
          product: {
            available: 0.0,
            code: '000000000091057659',
            name: 'PR108-20RIB VT2PRIB 80M BAG ELT500 N-B-E'
          },
          quantity: 1,
          salesOrderEntryNumber: '20',
          salesUnitOfMeasureCode: 'SSU'
        },
        {
          batchName: 'L42S7RMKP',
          deliveryItemNumber: '000010',
          product: {
            available: 0.0,
            code: '000000000088333144',
            name: '189-64VT2PRIB 50U BOX BAS500 B'
          },
          quantity: 10,
          salesOrderEntryNumber: '10',
          salesUnitOfMeasureCode: 'SSU'
        }
      ],
      farmer: {
        name: 'CARL GREEN',
        uid: '0009221009'
      },
      salesOrderId: 'BC-0002222577',
      salesYear: '2024',
      status: 'DELIVERED'
    }
    const { queryByText } = render(<DeliveryProductListMobile delivery={delivery} />)

    expect(queryByText('•')).toBeNull()
  })

  it('should not render interpunct if batchName is not  available', () => {
    const delivery: Delivery = {
      agent: {
        name: 'BRIAN HAVLIK',
        uid: '0009146406'
      },
      code: '0801760288',
      createdOnDateTime: '2024-06-24T13:23:14-05:00',
      entries: [
        {
          deliveryItemNumber: '000020',
          product: {
            available: 0.0,
            code: '000000000091057659',
            name: 'PR108-20RIB VT2PRIB 80M BAG ELT500 N-B-E'
          },
          quantity: 1,
          salesOrderEntryNumber: '20',
          salesUnitOfMeasureCode: 'SSU'
        },
        {
          deliveryItemNumber: '000010',
          product: {
            available: 0.0,
            code: '000000000088333144',
            name: '189-64VT2PRIB 50U BOX BAS500 B'
          },
          quantity: 10,
          salesOrderEntryNumber: '10',
          salesUnitOfMeasureCode: 'SSU',
          seedSize: 'R2'
        }
      ],
      farmer: {
        name: 'CARL GREEN',
        uid: '0009221009'
      },
      salesOrderId: 'BC-0002222577',
      salesYear: '2024',
      status: 'DELIVERED'
    }
    const { queryByText } = render(<DeliveryProductListMobile delivery={delivery} />)

    expect(queryByText('•')).toBeNull()
  })

  it('should render product quantity with units successfully', () => {
    const delivery: Delivery = {
      agent: {
        name: 'BRIAN HAVLIK',
        uid: '0009146406'
      },
      code: '0801760288',
      createdOnDateTime: '2024-06-24T13:23:14-05:00',
      entries: [
        {
          deliveryItemNumber: '000020',
          product: {
            available: 0.0,
            code: '000000000091057659',
            name: 'PR108-20RIB VT2PRIB 80M BAG ELT500 N-B-E'
          },
          quantity: 1,
          salesOrderEntryNumber: '20',
          salesUnitOfMeasureCode: 'SSU'
        },
        {
          batchName: 'L42S7RMKP',
          deliveryItemNumber: '000010',
          product: {
            available: 0.0,
            code: '000000000088333144',
            name: '189-64VT2PRIB 50U BOX BAS500 B'
          },
          quantity: 10,
          salesOrderEntryNumber: '10',
          salesUnitOfMeasureCode: 'SSU',
          seedSize: 'R2'
        }
      ],
      farmer: {
        name: 'CARL GREEN',
        uid: '0009221009'
      },
      salesOrderId: 'BC-0002222577',
      salesYear: '2024',
      status: 'DELIVERED'
    }
    const { getByText } = render(<DeliveryProductListMobile delivery={delivery} />)

    expect(getByText('1 SSU')).toBeTruthy()
    expect(getByText('10 SSU')).toBeTruthy()
  })
})
