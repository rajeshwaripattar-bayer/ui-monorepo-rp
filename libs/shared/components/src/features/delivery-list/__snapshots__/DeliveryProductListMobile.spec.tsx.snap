// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryProductListMobile should render (snapshot test) successfully 1`] = `
<body>
  <div>
    <div
      class="lmnt lmnt-card mdc-card mdc-card--outlined lmnt-card--left-top lmnt-theme-surface-bg card"
      style="height: auto;"
    >
      <div
        class="lmnt mdc-card__content"
      >
        <div
          class="lmnt lmnt-card-title lmnt-card-title--top-placement"
          data-testid="title"
        >
          <div
            class="lmnt-card-title__text"
          >
            <span
              class="mdc-typography lmnt mdc-typography--subtitle1"
            >
              deliveries.products_in_delivery.label
            </span>
          </div>
        </div>
        <div
          class="lmnt-card-body  mdc-typography--overline lmnt-theme-surface-bg card_body"
          data-testid="orderId"
        >
          orders.order.label 0002222577
        </div>
        <ul
          aria-multiselectable="false"
          aria-orientation="vertical"
          class="lmnt lmnt-list mdc-list lmnt-theme--bg product_list_mobile"
          data-evolution="true"
          role="listbox"
        >
          <div
            class="wrapper-list-item"
          >
            <li
              aria-checked="false"
              class="lmnt mdc-list-item lmnt-theme--bg mdc-list-item--with-two-lines lmnt-list-item__no-hover lmnt-list-item--wrap "
              role="option"
              tabindex="0"
            >
              <span
                class="mdc-list-item__ripple"
              />
              <span
                class="lmnt mdc-list-item__content"
              >
                <span
                  class="lmnt mdc-list-item__primary-text"
                >
                  PR108-20RIB VT2PRIB 80M BAG ELT500 N-B-E
                </span>
                <span
                  class="lmnt mdc-list-item__secondary-text"
                />
              </span>
              <span
                class="lmnt mdc-list-item__end"
              >
                <span
                  class="mdc-typography lmnt mdc-typography--caption"
                >
                  1 ST
                </span>
              </span>
            </li>
            <li
              class="mdc-list-divider lmnt-list-divider"
              role="separator"
            />
          </div>
          <div
            class="wrapper-list-item"
          >
            <li
              aria-checked="false"
              class="lmnt mdc-list-item lmnt-theme--bg mdc-list-item--with-two-lines lmnt-list-item__no-hover lmnt-list-item--wrap "
              role="option"
              tabindex="-1"
            >
              <span
                class="mdc-list-item__ripple"
              />
              <span
                class="lmnt mdc-list-item__content"
              >
                <span
                  class="lmnt mdc-list-item__primary-text"
                >
                  189-64VT2PRIB 50U BOX BAS500 B
                </span>
                <span
                  class="lmnt mdc-list-item__secondary-text"
                >
                  <span
                    class="mdc-typography lmnt mdc-typography--caption"
                    data-testid="batchName"
                  >
                    L42S7RMKP
                  </span>
                    â€¢
                  <span
                    class="mdc-typography lmnt mdc-typography--caption"
                    data-testid="seedSize"
                  >
                    R2
                  </span>
                </span>
              </span>
              <span
                class="lmnt mdc-list-item__end"
              >
                <span
                  class="mdc-typography lmnt mdc-typography--caption"
                >
                  1 SSU
                </span>
              </span>
            </li>
          </div>
        </ul>
      </div>
    </div>
  </div>
</body>
`;
